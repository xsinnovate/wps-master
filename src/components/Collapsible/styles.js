import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  button: {
    flexDirection: 'row',
    alignItems: 'center',
  },
  icon: {
    marginLeft: 10,
    fontSize: 32,
    fontWeight: '100',
    lineHeight: 32,
  },
  content: {
    paddingVertical: 10,
  },
});
