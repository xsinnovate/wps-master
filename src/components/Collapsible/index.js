import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import { Text, TouchableHighlight, View } from 'react-native';
import styles from './styles';

export default class Collapsible extends PureComponent {
  state = {
    isCollapsed: true,
  };

  onToggle = () => {
    this.setState(state => ({
      isCollapsed: !state.isCollapsed,
    }));
  }

  render() {
    const { isCollapsed } = this.state;
    const { children, label, labelHide } = this.props;

    return (
      <View>
        <TouchableHighlight
          activeOpacity={0.4}
          onPress={this.onToggle}
          underlayColor="#fff"
        >
          <View style={styles.button}>
            <Text>
              {!isCollapsed && labelHide ? labelHide : label}
            </Text>
            <Text style={styles.icon}>
              {!isCollapsed ? '-' : '+'}
            </Text>
          </View>
        </TouchableHighlight>
        {!isCollapsed ? (
          <View style={styles.content}>
            {children}
          </View>
        ) : null}
      </View>
    );
  }
}

Collapsible.defaultProps = {
  labelHide: '',
};

Collapsible.propTypes = {
  children: PropTypes.node.isRequired,
  label: PropTypes.string.isRequired,
  labelHide: PropTypes.string,
};
