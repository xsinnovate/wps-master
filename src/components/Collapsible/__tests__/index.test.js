import React from 'react';
import renderer from 'react-test-renderer';
import { TouchableHighlight } from 'react-native';
import Collapsible from '..';

describe('<Collapsible />', () => {
  it('renders correctly', () => {
    const tree = renderer.create((
      <Collapsible label="Label">
        Content
      </Collapsible>
    )).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('expands when label is pressed', () => {
    const rendered = renderer.create((
      <Collapsible label="Label">
        Content
      </Collapsible>
    ));
    const instance = rendered.root;

    instance.findByType(TouchableHighlight).props.onPress();

    expect(rendered.toJSON()).toMatchSnapshot();
  });

  it('displays labelHide prop when expanded', () => {
    const rendered = renderer.create((
      <Collapsible
        label="Label"
        labelHide="Hide"
      >
        Content
      </Collapsible>
    ));
    const instance = rendered.root;

    instance.findByType(TouchableHighlight).props.onPress();

    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
