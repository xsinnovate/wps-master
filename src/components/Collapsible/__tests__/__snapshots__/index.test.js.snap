// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Collapsible /> displays labelHide prop when expanded 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text>
        Hide
      </Text>
      <Text
        style={
          Object {
            "fontSize": 32,
            "fontWeight": "100",
            "lineHeight": 32,
            "marginLeft": 10,
          }
        }
      >
        -
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 10,
      }
    }
  >
    Content
  </View>
</View>
`;

exports[`<Collapsible /> expands when label is pressed 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text>
        Label
      </Text>
      <Text
        style={
          Object {
            "fontSize": 32,
            "fontWeight": "100",
            "lineHeight": 32,
            "marginLeft": 10,
          }
        }
      >
        -
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "paddingVertical": 10,
      }
    }
  >
    Content
  </View>
</View>
`;

exports[`<Collapsible /> renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Text>
        Label
      </Text>
      <Text
        style={
          Object {
            "fontSize": 32,
            "fontWeight": "100",
            "lineHeight": 32,
            "marginLeft": 10,
          }
        }
      >
        +
      </Text>
    </View>
  </View>
</View>
`;
